UnitsNet/QuantityFormatter.cs

```diff
@@ -206,19 +186,59 @@ private static string FormatUntrimmed<TUnitType>(IQuantity<TUnitType> quantity,
         // Anything else is a standard numeric format string with default unit abbreviation postfix.
         return FormatWithValueAndAbbreviation(quantity, format, formatProvider);
     }
-
-    private static string FormatWithValueAndAbbreviation<TUnitType>(IQuantity<TUnitType> quantity, string format, IFormatProvider formatProvider)
-        where TUnitType : struct, Enum
+    
+    private string FormatWithValueAndAbbreviation<TQuantity>(TQuantity quantity, string format, IFormatProvider formatProvider)
+        where TQuantity : IQuantity
     {
-        var abbreviation = UnitsNetSetup.Default.UnitAbbreviations.GetDefaultAbbreviation(quantity.Unit, formatProvider);
-        return string.Format(formatProvider, $"{{0:{format}}} {{1}}", quantity.Value, abbreviation);
+        var abbreviation = _unitAbbreviations.GetDefaultAbbreviation(quantity.UnitKey, formatProvider);
+        if (abbreviation.Length == 0)
+        {
+            return quantity.Value.ToString(format, formatProvider);
+        }
+
+#if NET
+        // TODO see about using the Span<char> overloads (net 8+)
+        return quantity.Value.ToString(format, formatProvider) + ' ' + abbreviation;
```

Comment:
No idea if it makes sense to bother with StringBuilder here, or if it's just overkill. I'm not familiar with any Span<char> overloads to concatenate strings without allocations, but maybe there exists some stuff here.

Reply:
it's more interesting than just the concatenation; the game is to use a `stacalloc` to create a buffer and then use `TryFormat`  from the `ISpanFormattable` interface on the number, appending the abbreviation after it- with zero intermediate strings allocations.